arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -315,7 +320,7 @@ RestStatus RestAdminClusterHandler::execute() {
 RestAdminClusterHandler::futureVoid RestAdminClusterHandler::retryTryDeleteServer(std::unique_ptr<RemoveServerContext>&& ctx) {
   if (++ctx->tries < 60) {
     return SchedulerFeature::SCHEDULER->delay(1s).thenValue(
-      [this, self = shared_from_this(), ctx = std::move(ctx)] (auto) mutable {
+      [this, ctx = std::move(ctx)] (auto) mutable {
```

Comment:
if `self` is not captured, `this` may be a dangling pointer when the callback is executed by the Scheduler...?

Reply:
This lambda is executed (or destroyed) strictly before the `thenFinal` lambda in `waitForFuture` which captures `self`.